src/controllers/demandcontroller.js

```diff
@@ -0,0 +1,69 @@
+class DemandController extends Controller
+{
+    constructor()
+    {
+        super();
+
+        this.yearDays = GAME.model.config.yearDays;
+        this.hours = GAME.model.config.hours;
+    }
+
+    doCustomerOrderGeneration()
+    {
+        // TODO every day, and every once in a while (structural and variable?). We need to think about this.
+        // Random component - About 1 per day = ~1.72, 2 per day = ~1.38.
+        if (this._normalDistribution() > 1.72 || GAME.model.config.hours % 24 == 8) {
+            const customerController = new CustomerController();
+            customerController.generateOrder();
+        }
+    }
+    /**
+     * Returns a measure of demand for each product
+     */
+    randomDemandGenerator(demand)
+    {
+        const mean = demand.mean + this._seasonalComponent(demand.seasonality);
+
+        return Math.max(
+            this._normalDistribution(mean, demand.variance),
+            0
+        );
+    }
+
+    /**
+     * Computes a seasonal modifier for specific demand.
+     *
+     * @param {Object} seasonality
+     * @private
+     */
+    _seasonalComponent(seasonality)
```

Comment:
This builds a seasonal component, assuming a certain period (set in config by `yearDays`) before repetition. The amplitude indicates the maximum increase/decrease against the equilibrium (`demand.mean`). A modifier is used for products that peak in the winter, as set by `isWinterProduct`.